samples/web/content/apprtc/js/appcontroller.js

```diff
@@ -119,28 +127,69 @@ var AppController = function(loadingParams) {
     this.call_.oncallerstarted = this.displaySharingInfo_.bind(this);
 
-    this.call_.start(this.loadingParams_.roomId);
-
-    window.onbeforeunload = this.call_.hangup.bind(this.call_);
-    document.onkeypress = this.onKeyPress_.bind(this);
-    window.onmousemove = this.showIcons_.bind(this);
-    
-    $(UI_CONSTANTS.muteAudioSvg).onclick = this.toggleAudioMute_.bind(this);
-    $(UI_CONSTANTS.muteVideoSvg).onclick = this.toggleVideoMute_.bind(this);
-    $(UI_CONSTANTS.fullscreenSvg).onclick = this.toggleFullScreen_.bind(this);
-    $(UI_CONSTANTS.hangupSvg).onclick = this.hangup_.bind(this);
-  
-    setUpFullScreen();
+    this.roomSelection_ = null;
+    if (this.loadingParams_.connect !== false && this.loadingParams_.roomId) {
+      // If we already have a room to connect to, connect now.
+      this.call_.start(this.loadingParams_.roomId);
+      
+      // Record this room in the recently used list.
+      var recentlyUsedList = new RoomSelection.RecentlyUsedList();
+      recentlyUsedList.pushRecentRoom(this.loadingParams_.roomId);
+      
+      this.finishCallSetup_();
+    } else {
+      // Display the room selection UI.
+      var roomSelectionDiv = $(UI_CONSTANTS.roomSelectionDiv);
+      this.roomSelection_ = new RoomSelection(roomSelectionDiv, this.loadingParams_.suggestedRoomId);
+      this.show_(roomSelectionDiv);
+      this.roomSelection_.onRoomSelected = function(roomName) {
+        this.hide_(roomSelectionDiv);
+        this.call_.start(roomName);
+        this.finishCallSetup_();
+      }.bind(this);
+    }
   }.bind(this)).catch(function(error) {
     trace('Error initializing: ' + error.message);
   }.bind(this));
 };
 
+AppController.prototype.finishCallSetup_ = function() {
+  this.inCall_ = true;
+  
+  window.onbeforeunload = this.call_.hangup.bind(this.call_);
+  document.onkeypress = this.onKeyPress_.bind(this);
+  window.onmousemove = this.showIcons_.bind(this);
+  
+  $(UI_CONSTANTS.muteAudioSvg).onclick = this.toggleAudioMute_.bind(this);
+  $(UI_CONSTANTS.muteVideoSvg).onclick = this.toggleVideoMute_.bind(this);
+  $(UI_CONSTANTS.fullscreenSvg).onclick = this.toggleFullScreen_.bind(this);
+  $(UI_CONSTANTS.hangupSvg).onclick = this.hangup_.bind(this);
+
+  setUpFullScreen();
+  
+  if (!isChromeApp()) {
+    window.onpopstate = function(event) {
+      if (!event.state) {
+        // Resetting back to room selection page not yet supported, reload
+        // the initial page instead.
+        trace('Reloading main page.');
+        location.href = location.origin;
+      } else {
+        // This could be a forward request to open a room again
+        if (event.state && event.state.roomLink) {
+          location.href = event.state.roomLink;
+        }
+      }
+    };
+  }
+};
+
 AppController.prototype.hangup_ = function() {
   trace('Hanging up.');
   this.hide_(this.icons_);
   this.displayStatus_('Hanging up');
   this.transitionToDone_();
+  this.inCall_ = false;
```

Comment:
It does not currently. I think our next step as we move to the more advanced UI is to plan out/clean up how we enter/exit calls so we can do it multiple times without completely reloading the page. This will be required for the Chrome app.